src/Yaml/test.zig

```diff
@@ -708,6 +708,31 @@ test "stringify a list" {
     try testStringify("[ 1, 2, 3 ]", arr);
 }
 
+test "pointer of a value" {
+    const TestStruct = struct {
+        a: usize,
+        b: i64,
+        c: u12,
+    };
+
+    const source =
+        \\a: 1
+        \\b: 2
+        \\c: 3
+    ;
+
+    var arena = std.heap.ArenaAllocator.init(testing.allocator);
+    defer arena.deinit();
+
+    var yaml = Yaml{ .source = source };
+    try yaml.load(arena.allocator());
+
+    const parsed = try yaml.parse(arena.allocator(), *TestStruct);
```

Comment:
I am looking at this again and I cannot help but wonder, what is the use case here? Why would I want to deserialize into a pointer? Does it have a use case that is more than just a shorthand for:

```zig
const parsed = try yaml.parse(arena.allocator(), TestStruct);
const ptr_to_parsed = &parsed;
```

Reply:
Recursive structures in a tagged union is ege expected use case.

Reply:
Mhm, that's a good example. Perhaps we should include that as a test (and example) rather than what we currently have proposed in this PR? Wdyt?

Reply:
Would you mind adding it so that we can land this PR?